### PR TITLE
Added microsite FAQs accordion component

### PR DESCRIPTION
**Pull request by rb1193**
An initial implementation of the FAQs accordion as per https://trello.com/c/VgcC4scz/29-build-module-components-for-microsites

The accordion doesn't animate the height transition (which I think is what the video prototype suggested it should do), instead it immediately resizes and fades in with opacity. However so far as I was able to find there isn't a particularly good way of animating the height without fixing it at some point, which I didn't think was appropriate. I could try something like [this](https://github.com/FredrikOseberg/react-anim-kit/blob/master/src/AnimateHeightContainer/AnimateHeightContainer.jsx) to get around it but I thought that was more trouble than its worth and probably going to feel janky.
___
NB - by making a pull request you confirm agreement to the [Contributors Agreement](https://github.com/openlawnz/openlawnz-web/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rb1193** [comment]
@BenRauzi @williamparry I think this should be good to go now

File: content/microsites/test-microsite-v2.json [hunk 1]
```diff
@@ -80,6 +80,24 @@
...
+              "answer_html": "This is the answer to the second frequently asked question"
+            },
+            {
+              "question": "Third FAQ",
+              "answer_html": "The answer to the third FAQ"
+            }
+          ]
         }
       ],
       "asd": "asd"
```
Comment (williamparry) [inline content/microsites/test-microsite-v2.json:103]: ?

File: gatsby-node.js [hunk 2]
```diff
@@ -63,7 +63,9 @@ exports.createPages = async ({graphql, actions: { createPage } }) => {
                 type
                 title
                 content {
-                  content_html
+                  content_html,
+                  question,
+                  answer_html
```
Comment (williamparry) [inline gatsby-node.js:68]: Not sure the content or answers have to be HTML. Can we use plain text or an array of plain text?

File: src/components/Accordion.jsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+              tabIndex="0"
+              aria-expanded={accordionState[index] === STATE_OPEN}
+              aria-controls={`accordion-item-container-${index}`}
+              onClick={() => toggleAccordionItem(index)}
+            >
+              <span className="accordion-item-toggle" />
+              <span className="accordion-item-title">{item.title}</span>
+            </button>
+            <div
+              id={`accordion-item-container-${index}`}
```
Comment (williamparry) [inline src/components/Accordion.jsx:39]: This id won't be unique if there are more than one instances on the page.

File: src/templates/microsite.js [hunk 4]
```diff
@@ -25,6 +26,15 @@ const Microsite = ({ pageContext }) => {
             }
           </div>
         )
+      case "faqs":
+        return (
+          <div key={idx} name={title}>
```
Comment (williamparry) [inline src/templates/microsite.js:31]: This key needs to be revised

File: src/components/Accordion.jsx (hunk 3 above)
Comment (rb1193) [reply src/components/Accordion.jsx]: I've created an id prop for the accordion now which these ids are based off of, so I think this is resolved in the latest commit

File: gatsby-node.js (hunk 2 above)
Comment (rb1193) [reply gatsby-node.js]: Using the plain text widget now

File: src/templates/microsite.js (hunk 4 above)
Comment (rb1193) [reply src/templates/microsite.js]: Is it ok if we handle this in a separate PR? It affects the other modules as well. I've been chatting with Ben, and I think we might be able to use the new preSave hook to create a unique ID for each "module" without exposing to the CMS, but I feel it's outside the scope of this PR

**Comment by rb1193** [comment]
I've updated this, would you mind having a second look @BenRauzi @williamparry ?

**Pull request merged**